src/sync/app_credentials.cpp

```diff
@@ -48,6 +51,12 @@ IdentityProvider provider_type_from_enum(AuthProvider provider)
             return IdentityProviderCustom;
         case AuthProvider::USERNAME_PASSWORD:
             return IdentityProviderUsernamePassword;
+        case AuthProvider::FUNCTION:
+            return IdentityProviderFunction;
+        case AuthProvider::USER_API_KEY:
+            return IdentityProviderUserAPIKey;
+        case AuthProvider::SERVER_API_KEY:
```

Comment:
Don't think I have seen this Auth Provider documented anywhere? Is that something we should expose in the SDK's?

Reply:
The Swift Stitch SDK has this, afaik its a valid credential and we should expose it.

Reply:
Sure, I have no problem with that. I just wasn't aware it was a thing since it isn't documented on the Stitch Website as far as I can tell.

Reply:
What is this AuthProvider?
(As a principle, we should not blindly expose everything the existing Stitch API has. Only if it has big value)

Reply:
That principle goes precisely against the original arrangement with the product team. 

Either way, this auth provider allows for logging in with a server generated API key. Which is useful for IoT and debugging.

Reply:
@bmunkholm , I've only stated the above to be clear that we should move forward on our current path, and then re-examine the value of features later on. I don't feel that we have time to get caught up in value add discussions right now.

Reply:
@jsflax But if it's a feature that we don't even document I assume it's not being used by users? It must be faster to not add it to all SDKs than adding it though?
It's likely fine to have in OS for now though for the sake of this PR.